include/items.php

```diff
@@ -845,15 +846,26 @@ function item_store($arr,$force_parent = false, $notify = false, $dontcache = fa
 		}
 	} else {
 		// This can happen - for example - if there are locking timeouts.
-		logger("Item wasn't stored - we quit here.");
-		q("COMMIT");
+		q("ROLLBACK");
+
+		// Store the data into a spool file so that we can try again later.
```

Comment:
I'm not sure if you really need a deep look into the current functions. I would need some skeleton, then I could do the work. I can write down which functions are needed in general.

Important is the fetching. I would like to replace the various database queries at many places with function calls.